api/server/router/volume/volume_routes.go

```diff
@@ -45,6 +48,9 @@ func (v *volumeRouter) postVolumesCreate(ctx context.Context, w http.ResponseWri
 
 	var req volumetypes.VolumesCreateBody
 	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		if err == io.EOF {
+			return errdefs.InvalidParameter(errors.New("no body provided in request"))
```

Comment:
"unexpected EOF while decoding request body"

Reply:
Yeah, it was an attempt to make the message more "readable", but perhaps it's okay to skip that; I'll update

Reply:
Actually, there's two different error types; unexpected EOF (e.g. incomplete JSON), and EOF (empty body)

Reply:
Incomplete JSON would be a syntax error from the json decoder.

`io.UnexpectedEOF` would be due to the decoder having read a full object + "extra", the "extra" getting stored in the buffer and then trying to read more and not being able to even complete the data in the buffer due to EOF.

Perhaps simply "got EOF will reading request body" is enough? I'd worry about tying this to the json decoder implementation, which doesn't really seem to document the EOF here.

Reply:
Yes, sounds good to me; updated 👍